arch/x86/zefi/zefi.c

```diff
@@ -140,7 +140,7 @@ uintptr_t __abi efi_entry(void *img_handle, struct efi_system_table *sys_tab)
 
 	efi_prepare_boot_arg();
 
-	for (int i = 0; i < sizeof(zefi_zsegs)/sizeof(zefi_zsegs[0]); i++) {
+	for (int i = 0; i < (ARRAY_SIZE(zefi_zsegs)/sizeof(zefi_zsegs[0])); i++) {
```

Comment:
This is not correct. The `ARRAY_SIZE()` macro itself does the division by `array[0]`, so you're now doing it twice here, which in turn will give the wrong result.

Reply:
Does it even work?  This isn't Zephyr code, it's an EFI bootloader for Zephyr and last I checked (I wrote it originally, but it's been a **long** time) it isn't building against the Zephyr include directories.